### ISSUE TITLE
test_reporterDeprecations fails on Windows Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8752]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8752|
|Type|enhancement|
|Created|2016-08-12 23:17:30Z|
|Branch|https://github.com/twisted/twisted/tree/8752-rodrigc-runner-py3|

On Windows with Python 3.5, if I do:

```
python bin\trial --reporter=bwverbose twisted.trial.test.test_runner.RunnerDeprecationTests.test_reporterDeprecations
```

I get:

```
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\twisted\trial\test\test_runner.py", line 1039, in test_reporterDeprecations
    __file__, f)
  File "C:\Users\Craig\twisted3\twisted\trial\_synctest.py", line 602, in assertWarns
    self.fail("Can't handle different warnings")
twisted.trial.unittest.FailTest: Can't handle different warnings

twisted.trial.test.test_runner.RunnerDeprecationTests.test_reporterDeprecations
-------------------------------------------------------------------------------
```

<details><summary>Searchable metadata</summary>

```
trac-id__8752 8752
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8752_rodrigc_runner_py3 8752-rodrigc-runner-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1471043850302075 1471043850302075
changetime__1471145857069932 1471145857069932
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

On Windows Python 2, the output of the DeprecationWarning for this test is:

```
twisted.trial.test.test_runner.FakeReporter should implement done() but doesn't. Falling back to printErrors() and friends. c:\Users\Craig\twisted3\twisted\trial\test\test_runner.py <function f at 0x00000000047CA048>
```

and on Windows Python 3, the output of the DeprecationWarning for this test is:

```
twisted.trial.test.test_runner.FakeReporter should implement done() but doesn't. Falling back to printErrors() and friends. C:\Users\Craig\twisted3\twisted\trial\test\test_runner.py <function RunnerDeprecationTests.test_reporterDeprecations.<locals>.f at 0x0000023C0E36C950>
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

In changeset 5b1fb457bc38f4e03863862502bf1d3a5c5ab1ce
```
#!CommitTicketReference repository="" revision="5b1fb457bc38f4e03863862502bf1d3a5c5ab1ce"
Add topfile for #8752
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/491

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_4"></a>|@markrwilliams commented|
|-|-|

Replying to [rodrigc](#note_1):
> On Windows Python 2, the output of the DeprecationWarning for this test is:
> 
> ```
> twisted.trial.test.test_runner.FakeReporter should implement done() but doesn't. Falling back to printErrors() and friends. c:\Users\Craig\twisted3\twisted\trial\test\test_runner.py <function f at 0x00000000047CA048>
> ```
> 
> and on Windows Python 3, the output of the DeprecationWarning for this test is:
> 
> ```
> twisted.trial.test.test_runner.FakeReporter should implement done() but doesn't. Falling back to printErrors() and friends. C:\Users\Craig\twisted3\twisted\trial\test\test_runner.py <function RunnerDeprecationTests.test_reporterDeprecations.<locals>.f at 0x0000023C0E36C950>
> ```

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_5"></a>|@markrwilliams commented|
|-|-|

I don't know why trac did that.  What I said:

That looks like reasonable output.  Please merge this.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc commented|
|-|-|

In changeset ce7b0279a2df913c074c6255acdb174d825c8dc6
```
#!CommitTicketReference repository="" revision="ce7b0279a2df913c074c6255acdb174d825c8dc6"
Add topfile for #8752
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 92d5333beab83fadae175d73ec7fcbf6b00a4d84
```
#!CommitTicketReference repository="" revision="92d5333beab83fadae175d73ec7fcbf6b00a4d84"
Merge pull request #491 from twisted/8752-rodrigc-runner-py3

Author: hawkowl, rodrigc
Reviewer: markrwilliams
Fixes: #8752

Fix test_reporterDeprecations on Windows Python 3
```
